src/Raven.Server/Documents/Handlers/AI/Agents/AbstractAiAgentProcessor.cs

```diff
@@ -9,47 +9,163 @@
+            await WriteResponseAsync(context, chatId, r);
         }
 
-        public (BlittableJsonReaderObject Parameter, string UserPrompt) GetStartChatOptions(BlittableJsonReaderObject obj)
+        public override async ValueTask ExecuteAsync()
         {
-            if (obj.TryGet(nameof(StartChatBody.Parameters), out BlittableJsonReaderObject parameters) == false)
-                throw new ArgumentException(nameof(StartChatBody.Parameters));
-            if (obj.TryGet(nameof(StartChatBody.Prompt), out string userPrompt) == false)
-                throw new ArgumentException("User prompt is missing");
+            using var token = RequestHandler.CreateHttpRequestBoundOperationToken();
+            var chatId = RequestHandler.GetStringQueryString("chatId", required: false);
+            var agent = RequestHandler.GetStringQueryString("id", required: false);
+
+            if (string.IsNullOrEmpty(chatId) && string.IsNullOrEmpty(agent))
+                throw new ArgumentException("Chat ID or agent name must be provided.");
+
+            if (string.IsNullOrEmpty(chatId) == false && string.IsNullOrEmpty(agent) == false)
+                throw new ArgumentException("Chat ID and agent name can't be provided together.");
+
+            using var _ = ContextPool.AllocateOperationContext(out DocumentsOperationContext context);
+            var body = await ReadRequestBodyAsync(context, token.Token);
 
-            return (parameters, userPrompt);
+            ChatDocument chatDocument = null;
+            AiAgentConfiguration configuration = null;
+
+            if (string.IsNullOrEmpty(chatId) == false)
+            {
+                using var __ = context.OpenReadTransaction();
+                var chat = RequestHandler.Database.DocumentsStorage.Get(context, chatId);
+                if (chat == null)
+                    throw new DocumentDoesNotExistException(chatId);
+
+                chatDocument = ChatDocument.ToDocument(chatId, chat.Data);
+                configuration = GetAiAgentConfiguration(chatDocument.Agent);
+            }
+
+            if (string.IsNullOrEmpty(agent) == false)
+            {
+                configuration = GetAiAgentConfiguration(agent);
+                chatDocument = new ChatDocument(agent, body.Parameters);
+                chatDocument.Initialize(context, configuration, body.UserPrompt);
+                chatId = BuildChatId(configuration);
+            }
+
+            await HandleRequest(context, configuration, chatId, chatDocument, body, token.Token);
         }
 
-        public async Task<(BlittableJsonReaderArray ActionResponse, string UserPrompt)> ReadResumeChatBodyAsync(JsonOperationContext context, CancellationToken token)
+        private string BuildChatId(AiAgentConfiguration configuration)
+        {
+            var agentPrefix = $"{configuration.Identifier}{RequestHandler.IdentityPartsSeparator}";
+            var collection = configuration.Persistence?.Collection ?? "Chats";
```

Comment:
this needs to go to contants, basically in the same place where `@all_docs` is defined

Reply:
Also our own collections should be named differently e.g. starting with `@`